message/validation/validation.go

```diff
@@ -262,6 +269,11 @@ func (mv *messageValidator) ValidatePubsubMessage(_ context.Context, peerID peer
 				}
 
 				mv.metrics.MessageRejected(valErr.Text(), descriptor.Role, round)
+				if errors.Is(err, ErrRSASignature) {
+					mv.peerRateLimiter.RegisterRSAErrorRequest(peerID)
+				} else {
+					mv.peerRateLimiter.RegisterInvalidRequest(peerID)
+				}
 				return pubsub.ValidationReject
 			}
```

Comment:
Increasing the invalid counter here, even though the message was rejected, was a good idea!!
Can you just add some comments explaining why we do that? Basically, to say that even though the message was rejected, it still consumes resources as a message that would be ignored.